pkg/build/controller/strategy/util.go

```diff
@@ -93,7 +93,7 @@ func mountSecretVolume(pod *kapi.Pod, secretName, mountPath string) {
 // allowing Docker to authenticate against private registries or Docker Hub.
 func setupDockerSecrets(pod *kapi.Pod, pushSecret, pullSecret *kapi.LocalObjectReference) {
 	if pushSecret != nil {
-		mountSecretVolume(pod, pushSecret.Name, DockerPushSecretMountPath)
+		mountSecretVolume(pod, pushSecret.Name, DockerPushSecretMountPath, "push-"+pushSecret.Name)
```

Comment:
Can length of "push-" + pushSecret.Name be longer than a valid volume name ? 
If so, it'd be good to use util.namer.GetName() here